File: src/main/java/de/nordgedanken/auto_hotkey/runconfig/core/AhkRunConfigFactory.java
package de.nordgedanken.auto_hotkey.runconfig.core;

import com.intellij.execution.configurations.ConfigurationFactory;
import com.intellij.execution.configurations.ConfigurationType;
import com.intellij.execution.configurations.RunConfiguration;
import com.intellij.openapi.project.Project;
import de.nordgedanken.auto_hotkey.util.AhkConstants;
import org.jetbrains.annotations.NotNull;

/**
 * Produces new Ahk run configurations
 */
public class AhkRunConfigFactory extends ConfigurationFactory {

	public AhkRunConfigFactory(ConfigurationType configurationType) {
		super(configurationType);
	}

	@Override
	public @NotNull RunConfiguration createTemplateConfiguration(@NotNull Project project) {
		return new AhkRunConfig(project, this, AhkConstants.LANGUAGE_NAME);
	}

	@Override
	public @NotNull String getId() {
		return AhkConstants.LANGUAGE_NAME;
	}
}

File: src/main/kotlin/de/nordgedanken/auto_hotkey/runconfig/core/AhkRunConfigFactory.kt
package de.nordgedanken.auto_hotkey.runconfig.core

import com.intellij.execution.configurations.ConfigurationFactory
import com.intellij.execution.configurations.RunConfiguration
import com.intellij.openapi.project.Project
import de.nordgedanken.auto_hotkey.util.AhkConstants

/**
 * Produces new Ahk run configurations
 */
class AhkRunConfigFactory(configurationType: AhkRunConfigType) : ConfigurationFactory(configurationType) {
    override fun createTemplateConfiguration(project: Project): RunConfiguration {
        return AhkRunConfig(project, this, AhkConstants.LANGUAGE_NAME)
    }

    override fun getId() = AhkConstants.LANGUAGE_NAME
}
